flink-ml-lib/src/main/java/org/apache/flink/ml/classification/linear/LogisticRegressionModel.java

```diff
@@ -0,0 +1,189 @@
+    }
+
+    public static LogisticRegressionModel load(StreamExecutionEnvironment env, String path)
+            throws IOException {
+        LogisticRegressionModel model = ReadWriteUtils.loadStageParam(path);
+        Table modelData =
+                ReadWriteUtils.loadModelData(
+                        env, path, LogisticRegressionModelData.getModelDataDecoder());
+        return model.setModelData(modelData);
+    }
+
+    @Override
+    public LogisticRegressionModel setModelData(Table... inputs) {
+        model = inputs[0];
+        return this;
+    }
+
+    @Override
+    public Table[] getModelData() {
+        return new Table[] {model};
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public Table[] transform(Table... inputs) {
+        Preconditions.checkArgument(inputs.length == 1);
+        StreamTableEnvironment tEnv =
+                (StreamTableEnvironment) ((TableImpl) inputs[0]).getTableEnvironment();
+        DataStream<Row> data = tEnv.toDataStream(inputs[0]);
+        final String broadcastModelKey = "broadcastModel";
+        DataStream<LogisticRegressionModelData> modelData =
+                LogisticRegressionModelData.getModelDataStream(model);
+        DataStream<Row> predictResult =
+                BroadcastUtils.withBroadcastStream(
+                        Collections.singletonList(data),
+                        Collections.singletonMap(broadcastModelKey, modelData),
+                        inputList -> {
+                            DataStream inputData = inputList.get(0);
+                            return inputData.transform(
+                                    "doPrediction",
+                                    new RowTypeInfo(
+                                            new TypeInformation[] {
+                                                TypeInformation.of(DenseVector.class),
+                                                BasicTypeInfo.DOUBLE_TYPE_INFO,
+                                                PrimitiveArrayTypeInfo
+                                                        .DOUBLE_PRIMITIVE_ARRAY_TYPE_INFO
+                                            },
+                                            new String[] {
+                                                getFeaturesCol(),
+                                                getPredictionCol(),
+                                                getRawPredictionCol()
+                                            }),
+                                    new PredictOp(
+                                            new PredictOneRecord(
+                                                    getFeaturesCol(), broadcastModelKey)));
+                        });
+
+        return new Table[] {tEnv.fromDataStream(predictResult)};
+    }
+
+    /** A utility operator used for prediction. */
+    private static class PredictOp extends AbstractUdfStreamOperator<Row, RichMapFunction<Row, Row>>
```

Comment:
There are already a couple places (e.g. `MapPartitionOperator`) where we specify the computation logic in `AbstractStreamOperator::processElement` instead of putting this logic in a separate `RichMapFunction` subclass. Could we do the same here for consistency and simplicity?

Reply:
We can only access the broadcast variables in `RichFunction` in the implementation of `withBroadcast`, so a `RichFunction` is inevitable.

Moreover, I do believe separate the logic for predicting one record is more clear.